### DISCUSSION TITLE
behavior at high count rates, > 7000 cps

### DISCUSSION THREAD

**Discussion by jbeale1** [Open-Gamma-Project/Gamma-MCA#229] [General]
Just for fun, I put my entire collection of sources under a 3" PMT + plastic scintillator, connected to the OpenGamma board. The PWA app reports over 7000 cps on average, but the instantaneous CPS report fluctuates in a fairly periodic way from under 4k to over 10k CPS. This may be due to USB congestion on my slow Linux box or other hardware limits, and this is not at all a complaint, just an example showing the combined system does at least reach this count rate.
![GammPWA_2024-02-04 15-56-38](https://github.com/OpenGammaProject/Gamma-MCA/assets/7452231/52f5b261-a059-44eb-9212-4c1f96303307)

**Comment by dc1rdb** [comment]
Similar effects here with my setup, using my portable OGD HW Rev. 4, FW 4.2.0. I'm running Gamma MCA on a Panasonic Toughpad, which is not really a high performance machine. So USB or hardware limits could definitely be an issue.
The effect does not bother me for my regular spectroscopy tasks, though.
![gamma_mca_evolution (1)](https://github.com/OpenGammaProject/Gamma-MCA/assets/9314230/190c6990-cc8d-43e5-840e-26c0688f1aa0)

**Comment by NuclearPhoenixx** [comment]
Yeah that's definitely somewhat attributed to USB congestion and a timing issue due to the transfer rate. You can try to switch to histogram mode both on the device itself (`set out spectrum`) and in the app ("Data Order" to the right). That significantly reduces the amount of data needed to be transferred above like 4kcps. If the bottleneck is more on the computer side then it's not gonna help either and maybe make things actually a little bit worse, you'll have to try 😉
